packages/flutter_tools/lib/src/commands/clean.dart

```diff
@@ -28,38 +32,70 @@ class CleanCommand extends FlutterCommand {
 
   @override
   Future<FlutterCommandResult> runCommand() async {
+    // Clean Xcode to remove intermediate DerivedData artifacts.
+    // Do this before removing ephemeral directory, which would delete the xcworkspace.
+    final FlutterProject flutterProject = FlutterProject.current();
```

Comment:
noob question, do we already know they're in a valid flutter project directory?

Reply:
`flutter clean` requires a pubspec.yaml
```
$ flutter clean

Error: No pubspec.yaml file found.
This command should be run from the root of your Flutter project.
Do not run this command from the root of your git clone of Flutter.
```